applications/nrf_desktop/doc/porting_guide.rst

```diff
@@ -1,10 +1,152 @@
+
+      * |preconfigured_build_types|
+
+Desktop mouse (nrf52_pca20044 and nrf52810_pca20045)
+      * Both boards are meant for the project-specific hardware and are defined in ``nrf/boards/arm/nrf52_pca20044`` and ``nrf/boards/arm/nrf52810_pca20045``, respectively.
+      * The application is configured to act as a mouse.
+      * Only the Bluetooth Low Energy transport is enabled.
+        Bluetooth uses Zephyr's software link layer.
+      * There is no configuration with bootloader available.
+
+
+Sample mouse or keyboard (nrf52840_pca10056)
+      * The configuration uses the nRF52840 Development Kit.
+      * The build types allow to build the application both as mouse and as keyboard.
+      * Inputs are simulated based on the hardware button presses.
+      * The configuration with bootloader is available.
+
+Keyboard (nrf52_pca20037)
+      * The board used is defined in ``nrf/boards/arm/nrf52_pca20037`` for the project-specific hardware.
+      * The application is configured to act as a keyboard, with the BLE transport enabled.
+      * Bluetooth is configured to use Nordic's proprietary link layer.
+      * |preconfigured_build_types|
+
+Dongle (nrf52840_pca10059)
+      * This configuration uses Nordic's nRF52840 dongle defined in Zephyr.
+      * Since the board is generic, project-specific changes are applied in the DTS overlay file.
+      * The application is configured to act as both mouse and keyboard.
+      * Bluetooth uses Nordic's proprietary link layer and is configured to act as a central.
+        Input data come from Bluetooth and are retransmitted to the USB.
+      * |preconfigured_build_types|
+
+.. _porting_guide_adding_board:
+
+Adding a new board
+******************
+
+When adding a new board for the first time, focus on a single configuration.
+Moreover, keep the default ``ZDebug`` build type that the application is built with, and do not add any additional build type parameters.
+
+.. note::
+    The following procedure will use the gaming mouse configuration as example.
+
+To use the nRF Desktop project with your custom board:
+
+1. Define the board by copying the nRF Desktop board files that are the closest match for your hardware.
+   For example, for gaming mouse use ``nrf/boards/arm/nrf52840_pca20041``.
+#. Edit the DTS files to make sure they match the hardware configuration.
+   Pay attention to pins that are used and to the bus configuration for optical sensor.
+#. Copy the project files for the device that is the closest match for your hardware.
+   For example, for gaming mouse these are located at ``applications/nrf_desktop/configure/nrf52840_pca20041``.
+#. Optionally, depending on the board, edit the DTS overlay file.
+   This step is not required if you have created a new board and its DTS files fully describe your hardware.
+   In such case, the overlay file can be left empty.
+#. In Kconfig, ensure that the following modules that are specific for gaming mouse are enabled:
+
+    * :ref:`motion`
```

Comment:
These IDs must be unique across the whole documentation set. I'd recommend adding some kind of prefix to them (e.g. `nrfdesktop_motion`) so that we don't run into problems with duplicated IDs at some point.

Reply:
True, I think this will have to be a wide change around the nrf_desktop doc files. Not only this single name can lead to problems. Maybe this could be done in subsequent PR.